### PR TITLE
HZN-973: Provide possibility to deploy ReST services inside Karaf Container

### PR DESCRIPTION
**Pull request by mvrueden**
* JIRA: https://issues.opennms.org/browse/HZN-973

Please squash-merge.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1351,6 +1351,7 @@
     <osgiEnterpriseVersion>5.0.0</osgiEnterpriseVersion>
     <owaspEncoderVersion>1.2.1</owaspEncoderVersion>
     <owaspHtmlSanitizerVersion>20170515.1</owaspHtmlSanitizerVersion>
+    <osgiJaxRsVersion>1.0.0-SNAPSHOT</osgiJaxRsVersion>
```
Comment (j-white) [inline pom.xml]: We should reference a released version before we merge this.

File: container/bridge/src/test/java/org/opennms/container/web/felix/base/internal/handler/ServletHandlerRequestTest.java [hunk 2]
```diff
@@ -92,11 +90,12 @@ public ServletHandlerRequestTest(final HttpServletRequest request, final String
     }
 
     @Test
+    // TODO MVR fix me
```
Comment (j-white) [inline container/bridge/src/test/java/org/opennms/container/web/felix/base/internal/handler/ServletHandlerRequestTest.java]: TODO

Comment (mvrueden) [reply container/bridge/src/test/java/org/opennms/container/web/felix/base/internal/handler/ServletHandlerRequestTest.java]: Yeah there is still some more work to do, I just wanted to have it as a PR so I can find it more easily (-:

**Comment by mvrueden** [comment]
I took care of the review comments.

File: smoke-test/src/test/java/org/opennms/smoketest/minion/SyslogKafkaElasticsearchBufferingIT.java [hunk 3]
```diff
@@ -63,6 +64,7 @@
  * 
  * @author Seth
  */
+@Ignore
```
Comment (j-white) [inline smoke-test/src/test/java/org/opennms/smoketest/minion/SyslogKafkaElasticsearchBufferingIT.java]: I think you may need to add the new `opennms-osgi-core-rest` feature to `https://github.com/OpenNMS/opennms/blob/1c8c7b8da26320fa2cc4ad0a764c72597bfc4fea/smoke-test/src/test/resources/org.apache.karaf.features.cfg#L87` to get this one to work.

**Comment by j-white** [comment]
👍 works well.

Being able to reload bundles exposing REST endpoints on the fly is quite nice.

**Comment by mvrueden** [comment]
Tests now pass. 🤦‍♂️ on not seeing the missing feature on the failing test (-:

**Pull request merged**